### PR TITLE
Tabular result format

### PR DESCRIPTION
**Pull request by lukasmalkmus**
A first iteration for the tabular results format, something we worked on to replace the current response format in the future. This draft serves as a place to discuss, test and iterate on the changes.

Three commits are of particular interest (omitting SHAs/links as we will likely rebase this):

1. Implements the general format
2. Provides a generic iterator for columns and rows
3. Adds stronger typing for ease of use and protection against misuse
4. Adds support for Go's 1.23 new `iter` package

### CODE REVIEW

File: axiom/query/field.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+		case TypeTimespan.String():
+			ft |= TypeTimespan
+		case TypeArray.String():
+			ft |= TypeArray
+		case TypeDictionary.String():
+			ft |= TypeDictionary
+		case TypeUnknown.String():
+			ft |= TypeUnknown
+		default:
+			return TypeInvalid, fmt.Errorf("unknown field type: %s", t)
```
[the remark this answers is not in the document]
Comment (lukasmalkmus) [reply axiom/query/field.go:55]: This piece of code is quite dated by now (when I initially designed it). This might go away altogether.

**Review by bahlo — APPROVED** [review]
Fancy iterators ✨

**Comment by kamstrup** [comment]
> Fancy iterators ✨

Only if it comes from the French region of Iteráteur; otherwise it's just sparkling for-loops

**Review by bahlo — APPROVED** [review]
cool

**Pull request merged**